### PR TITLE
Compare two submissions at segment level

### PR DESCRIPTION
**Pull request by snukky**
Adding a view with segment-level comparison of two submissions with highlighted token differences. This is an initial work for #12.

Changes proposed in the pull request:
- Added view `/compare/<submission_id>/<submission_id>`, not accessible from anywhere yet (type the URL).
  - Maybe we should use `/diff/<submission_id>/<submission_id>` and reserve `/compare` for comparing **systems**?
  - Token replacements, insertions and deletions are highlighted in light yellow, green, red.
  - Anonymous submissions made by other teams or submissions from different test sets cannot be compared.
- Added view `/submission/<submission_id>` showing source and hypothesis segments.
  - Clickable links to submissions on team and leaderboard pages (only for your own or public submissions).
  - A dropbox to select a submission to compare with.
- Segments on both views are paginated into up to 100 segments per page.
- Refactorized navbar and jumbotron blocks in templates to avoid copying.
- Added bootstrap CSS to forms.
- Added first docstrings to some model methods.
- Added unit tests.

Please notice that this PR is opened on top of #16.

@AppraiseDev/core-team

### CODE REVIEW

File: evaluation/apps.py [hunk 1]
```diff
@@ -0,0 +1,10 @@
+"""
+Project OCELoT: Open, Competitive Evaluation Leaderboard of Translations
+"""
+from django.apps import AppConfig
+
+
+class EvaluationConfig(AppConfig):
+    """ocelot evaluation app Configuration"""
```
Comment (cfedermann) [inline evaluation/apps.py:8]: OCELoT

Comment (cfedermann) [reply evaluation/apps.py:8]: `OCELoT evaluation app configuration`

File: evaluation/tests.py [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+from leaderboard.models import Competition
+from leaderboard.models import Language
+from leaderboard.models import SGML_FILE
+from leaderboard.models import Submission
+from leaderboard.models import Team
+from leaderboard.models import TestSet
+from leaderboard.models import TEXT_FILE
+from ocelot.settings import BASE_DIR
+
+TESTDATA_DIR = os.path.join(BASE_DIR, 'leaderboard/testdata')
```
Comment (cfedermann) [inline evaluation/tests.py:19]: @snukky: start using `pathlib.Path` instead? We are on Python 3 anyways, so this should be available. Makes paths much more readable...

File: evaluation/views.py [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+from django.shortcuts import HttpResponseRedirect
+from django.shortcuts import render
+
+from leaderboard.models import Submission
+from leaderboard.views import _get_team_data
+
+SEGMENTS_PER_PAGE = 100
+
+
+def _annotate_texts_with_span_diffs(text1, text2, char_based=False):
```
Comment (cfedermann) [inline evaluation/views.py:18]: @snukky: I really like this! 😍

File: ocelot/urls.py [hunk 4]
```diff
@@ -46,6 +50,9 @@
     path('teampage', teampage, name='teampage-view'),
     path('updates', updates, name='updates-view'),
     path('welcome', welcome, name='welcome-view'),
+    # evaluation app
+    path('compare/<sub_a_id>/<sub_b_id>', compare, name='compare-view'),
```
Comment (cfedermann) [inline ocelot/urls.py:54]: @snukky: use `'submission/<sub_a_id>/<sub_b_id>'` for comparison? No need for a specific `compare` view name?

**Review by cfedermann — CHANGES_REQUESTED** [review]
@snukky: we'll talk about this in today's call in 15 minutes. Thanks!

**Review by cfedermann — APPROVED** [review]
Good to go as well :)

File: evaluation/apps.py (hunk 1 above)
Comment (snukky) [reply evaluation/apps.py:8]: Fixed in both `leaderboard/apps.py` and `evaluation/apps.py`

File: evaluation/tests.py (hunk 2 above)
Comment (snukky) [reply evaluation/tests.py:19]: I agree that `pathlib` will be a better choice. I will open a new PR for this.

File: ocelot/urls.py (hunk 4 above)
Comment (snukky) [reply ocelot/urls.py:54]: That's a good idea! Changed as suggested.

**Pull request merged**